src/main/java/com/denizenscript/depenizen/bukkit/bridges/JobsBridge.java

```diff
@@ -1,70 +1,86 @@
 import com.denizenscript.denizen.objects.PlayerTag;
 import com.denizenscript.denizencore.objects.ObjectFetcher;
-import com.denizenscript.denizencore.tags.TagRunnable;
 import com.denizenscript.denizencore.objects.core.ListTag;
 import com.denizenscript.denizencore.objects.properties.PropertyParser;
-import com.denizenscript.denizencore.tags.ReplaceableTagEvent;
-import com.denizenscript.denizencore.tags.Attribute;
 import com.denizenscript.depenizen.bukkit.properties.jobs.JobsPlayerProperties;
 import com.denizenscript.denizencore.tags.TagManager;
 
 public class JobsBridge extends Bridge {
 
     @Override
     public void init() {
-        ObjectFetcher.registerWithObjectFetcher(JobsJobTag.class);
-        PropertyParser.registerProperty(JobPlayer.class, JobsJobTag.class);
+        /////////////////
+        // Register Events
+        /////////////////
+        ScriptEvent.registerScriptEvent(JobsJobsPaymentScriptEvent.class);
+        ScriptEvent.registerScriptEvent(JobsJobsExpGainScriptEvent.class);
+        ScriptEvent.registerScriptEvent(JobsJobsJoinScriptEvent.class);
+        ScriptEvent.registerScriptEvent(JobsJobsLeaveScriptEvent.class);
+        ScriptEvent.registerScriptEvent(JobsJobsLevelUpScriptEvent.class);
+        /////////////////
+        // Register Tag and Properties
+        /////////////////
+        ObjectFetcher.registerWithObjectFetcher(JobsJobTag.class, JobsJobTag.tagProcessor);
         PropertyParser.registerProperty(JobsPlayerProperties.class, PlayerTag.class);
-        TagManager.registerTagHandler(new TagRunnable.RootForm() {
-            @Override
-            public void run(ReplaceableTagEvent event) {
-                tagEvent(event);
-            }
-        }, "jobs");
-        DenizenCore.commandRegistry.registerCommand(JobsCommand.class);
-    }
+        PropertyParser.registerProperty(JobsJobProperties.class, JobsJobTag.class);
 
-    public void tagEvent(ReplaceableTagEvent event) {
-        Attribute attribute = event.getAttributes();
-        JobsJobTag j = null;
-        if (attribute.hasParam()) {
-            // Documented below.
-            if (JobsJobTag.matches(attribute.getParam())) {
-                j = attribute.paramAsType(JobsJobTag.class);
+        /////////////////
+        // Register root tag handler
+        /////////////////
+
+        // <--[tag]
+        // @attribute <jobs[<name>]>
+        // @returns JobsJobTag
+        // @plugin Depenizen, Jobs
+        // @description
+        // Returns the job tag with the given name
+        // -->
+        TagManager.registerTagHandler(ObjectTag.class, "jobs", (attribute) -> {
+            if (attribute.hasParam()) {
+                return JobsJobTag.valueOf(attribute.getParam(), attribute.context);
             }
-            else {
-                attribute.echoError("Could not match '" + attribute.getParam() + "' to a valid job!");
-                return;
+            // <--[tag]
+            // @attribute <jobs.server_jobs>
```

Comment:
This makes a really dirty/messy tag - if `jobs` is `JobsJobTag`, then `jobs.server_jobs` is ... `JobsJobTag.server_jobs`, but it isn't, which...

Split these up.

Reply:
As long as you don't mind removing `<jobs>` immediately, then we can do that.
This change was solely to maintain the existing behaviour of `<jobs>` returning a ListTag of all of the server jobs

Reply:
This remains unresolved.